corehq/motech/generic_inbound/reports.py

```diff
@@ -0,0 +1,100 @@
+from corehq.apps.reports.filters.base import BaseMultipleOptionFilter, BaseSimpleFilter
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.reports.standard import DatespanMixin
+from corehq.toggles import GENERIC_INBOUND_API
+
+from .models import RequestLog, ProcessingAttempt
+
+
+class RequestStatusFilter(BaseMultipleOptionFilter):
+    slug = 'request_status'
+    label = gettext_lazy('Request Status')
+
+    @property
+    def options(self):
+        return RequestLog.Status.choices
+
+
+class FormIdFilter(BaseSimpleFilter):
+    slug = "form_id"
+    label = gettext_lazy("Form ID")
+    help_inline = gettext_lazy("Enter a form id to filter results")
+
+
+class ApiRequestLogReport(DatespanMixin, GenericTabularReport):
+    name = gettext_lazy('Inbound API Request Logs')
+    slug = 'api_request_log_report'
+    base_template = "reports/base_template.html"
+    section_name = gettext_lazy('Project Settings')
+    dispatcher = DomainReportDispatcher
+    ajax_pagination = True
+    sortable = False
+
+    fields = [
+        'corehq.apps.reports.filters.dates.DatespanFilter',
+        'corehq.motech.generic_inbound.reports.RequestStatusFilter',
+        'corehq.motech.generic_inbound.reports.FormIdFilter',
+    ]
+
+    toggles = [GENERIC_INBOUND_API]
+
+    @classmethod
+    def allow_access(cls, request):
+        return request.couch_user.is_domain_admin()
+
+    @property
+    def total_records(self):
+        return self._queryset.count()
+
+    @property
+    def shared_pagination_GET_params(self):
+        return [
+            {'name': param, 'value': self.request.GET.getlist(param)}
+            for param in ['request_status', 'startdate', 'enddate', 'form_id']
+        ]
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(_("API")),
+            DataTablesColumn(_("Timestamp")),
+            DataTablesColumn(_("Status")),
+            DataTablesColumn(_("Response")),
+            DataTablesColumn(_("Form ID")),
+        )
+
+    @cached_property
+    def _queryset(self):
+        queryset = RequestLog.objects.filter(
+            domain=self.domain,
+            timestamp__gte=self.request.datespan.startdate_utc,
+            timestamp__lt=self.request.datespan.enddate_utc,
+        )
+        status = self.request.GET.getlist('request_status')
+        if status:
+            queryset = queryset.filter(status__in=status)
+        input_form_id = self.request.GET.get('form_id')
+        if input_form_id:
+            attempts = ProcessingAttempt.objects.filter(xform_id=input_form_id)
+            queryset = queryset.filter(id__in=[attempt.log for attempt in attempts])
+        return queryset
+
+    @property
+    def rows(self):
+        status_labels = dict(RequestLog.Status.choices)
+        for log in self._queryset[self.pagination.start:self.pagination.end]:
+            processing_attempt = log.processingattempt_set.get(log=log.id)
```

Comment:
This could return multiple attempts. Also, you should be able to adjust the main query to fetch the xform ID from the last attempt by using a subquery: https://docs.djangoproject.com/en/4.1/ref/models/expressions/#subquery-expressions

If we do that then the `total_records` query should not include the subquery since that's more expensive.

I'm not sure which is more efficient given that the query is paginated, but another approach would be to use `prefetch_related`: https://docs.djangoproject.com/en/4.1/ref/models/querysets/#prefetch-related

This may be better since we generally expect there to be only one attempt anyway.

Reply:
Thinking about this again, I'm not sure we even need to show the xform ID on the report at all. It's probably not useful to show in the report since it's just a GUID. If we ant to link to it we can make a request that redirects to the form.

Reply:
In that case I'll remove the Form ID column and, as per the above comment, add filtering by API.